### ISSUE TITLE
SauceLabs iOS 11.3 and 12: "message": "An unknown server-side error occurred while processing the command. Original error: Recursive object cannot be transferred"

### ISSUE THREAD

**Issue by janpio** [apache/cordova-paramedic#119]
When adding iOS 11.3 and 12 to the CI/SauceLabs setup, https://github.com/apache/cordova-paramedic/pull/118/files, the builds/tests fail:
https://travis-ci.org/apache/cordova-paramedic/builds/516361578

The 2 relevant failures are:
https://travis-ci.org/apache/cordova-paramedic/jobs/516361585
https://travis-ci.org/apache/cordova-paramedic/jobs/516361586

When looking at the output at Travis, you see this:

```
...
cordova-paramedic: waiting for test results
appium: Error: [execute(null,[])] Error response status: 13, , UnknownError - An unknown server-side error occurred while processing the command. Selenium error: An unknown server-side error occurred while processing the command. Original error: Recursive object cannot be transferred
cordova-paramedic: Tests failed to complete; ending appium session. The error is:
Error: waitForTests: Seems like device not connected to local server in 540 secs
    at Timeout._onTimeout (/Users/travis/build/apache/cordova-paramedic/lib/paramedic.js:368:24)
    at ontimeout (timers.js:386:11)
    at tryOnTimeout (timers.js:250:5)
    at Timer.listOnTimeout (timers.js:214:5)
From previous event:
    at ParamedicSauceLabs.runSauceTests (/Users/travis/build/apache/cordova-paramedic/lib/ParamedicSauceLabs.js:440:6)
    at ParamedicRunner.runTests (/Users/travis/build/apache/cordova-paramedic/lib/paramedic.js:338:40)
    at /Users/travis/build/apache/cordova-paramedic/lib/paramedic.js:95:21
Start running Appium tests...
Running Appium tests on Sauce Labs
Couldn't find Appium tests, skipping...
Completed tests at 8:32:46 PM
...
```

If you open the SauceLabs test run (URL at the end of the Travis output), you see that the tests actually run as expected, but only the transferring of the result then fails:

https://app.saucelabs.com/tests/32cb9379e1014dcda8555ebf3bfbe46f

```
  {
    "screenshot": 1,
    "between_commands": 0.24466705322265625,
    "start_time": 1554495831.065799,
    "suggestion_values": [],
    "request": {
      "args": [],
      "script": "return (function () {\n        if (typeof window._jasmineParamedicProxyCache === 'undefined') {\n            // wrong window\n            return null;\n        }\n        // get the results and clean up the cache\n        var result = window._jasmineParamedicProxyCache;\n        window._jasmineParamedicProxyCache = [];\n        return result;\n    }).apply(null, arguments);"
    },
    "HTTPStatus": 500,
    "result": {
      "message": "An unknown server-side error occurred while processing the command. Original error: Recursive object cannot be transferred"
    },
    "suggestion": null,
    "duration": 0.08858108520507812,
    "path": "execute",
    "in_video_timeline": 69.76390409469604,
    "method": "POST",
    "statusCode": 13
  },
```

After this, our script goes into an endless loop of trying to close a modal that is not there and then repeating this failing step.

**Comment by timbru31** [comment]
To shed some light into this issue here:

- it's an error thrown by Selenium: https://github.com/SeleniumHQ/selenium/commit/0fd840096fec20363daef3161a825e937b8bdc7e
- it's correct to be thrown as per W3C spec: https://w3c.github.io/webdriver/#dfn-internal-json-clone-algorithm (see "Otherwise")

Why does it happen?

- Because Appium >= 1.8.1 is used for iOS 11 and iOS 12.
- Since Appium 1.8.0 is fully compliant to the W3C WebDriver spec (https://github.com/appium/appium/blob/master/CHANGELOG.md#changes-in-version-180-from-172) the following `execute` is the culprit:
<img width="574" alt="Screen Shot 2019-04-11 at 00 49 55" src="https://user-images.githubusercontent.com/838003/55918533-cada5e80-5bf3-11e9-869b-fd41569c20d7.png">

- This custom JS injected into the client is actually from cordova-paramedic and part of the polling: https://github.com/apache/cordova-paramedic/blob/master/lib/appium/helpers/wdHelper.js#L229-L238


---

TL;DR: We need to update the polling algorithm in order to be W3C compliant.
*sidenote: only assumptions, I might have looked at the wrong places, too*

**Comment by janpio** [comment]
No, that looks right - that code is definitely the one throwing that response. Whatever object this returns seems to be considered as "recursive object" - which unfortunately is where I got totally lost and don't understand what is causing this and what is going on.

**Comment by janpio** [comment]
Probably solved via https://github.com/apache/cordova-paramedic/pull/118/commits/f5e4b2450833dc699a91b3db4d5c8eeb2df52880
